### ISSUE TITLE
Regression: ompgsql loses one message when Postgres isn't available

### ISSUE THREAD

**Issue by radu-gheorghe** [rsyslog/rsyslog#1105]
If a queue is defined, and there are proper retry settings, ompgsql doesn't seem to lose data in rsyslog 7.4.4 if PostgreSQL is down for a while. With latest rsyslog versions (8.19+) it seems to lose the first message.

Here's a sample config (though I believe a pure memory queue is enough to reproduce):

```
module(load="imuxsock")
module(load="ompgsql")

$ActionQueueType FixedArray
$ActionQueueSaveOnShutdown on
$ActionQueueFileName action_spillover
$WorkDirectory /var/run
$ActionQueueHighWaterMark 5000
$ActionQueueLowWaterMark 2000
$ActionQueueMaxDiskSpace 100m
$ActionQueueSize 100000
$ActionResumeRetryCount 5
$ActionResumeInterval 10
*.* :ompgsql:server,database,user,pass
```

With this config, the reproducing steps would be:
- start rsyslog, make sure messages end up in Postgres
- stop Postgres
- send a few messages
- start Postgres again, while rsyslog is still trying (before resumeInterval*resumeRetryCount)

Here, in 7.4.4 I would get all messages back, while in 8.19+ I get them all but one.

I've checked with other modules, like:
- omfwd via TCP: some messages are lost, I assume due to OS network buffers
- omelasticsearch: no message is lost, but I thought it may have something to do with the new config syntax
- ommysql: no message is lost. At which point I concluded it's an ompgsql bug. Am I wrong?

If not, then I looked over the ompgsql commits since 7.4.4 and on the surface it seems like this is the (only?) likely candidate: https://github.com/rsyslog/rsyslog/commit/ccca729c550466fa1f55fe8d1b2391489251ace8

I'll continue digging, but any feedback here would be helpful.

Also, this has nothing to do with https://github.com/rsyslog/rsyslog/pull/1087 - the bug appears with and without the patch.

**Comment by radu-gheorghe** [comment]
Interesting, I also see this in the postgres log:

`WARNING:  there is already a transaction in progress`

Not sure if this helps or not.

**Comment by radu-gheorghe** [comment]
Another interesting data point: with omlibdbi I can write to Postgres and no message is lost under the same circumstances.

**Comment by Asmerok** [comment]
I can confirm this. Easily reproducible with 8.19.

**Comment by rgerhards** [comment]
This is most probably resolved in at least 8.30, probably also some earlier versions. I remember fixing the exact same problem for other modules inside rsyslog core. If someone still experiences it, please let us know and I can re-open (or simply create a new issue). Thx!
